galasa-ui/src/utils/3270/getArrayOfImageCharacters.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import {
+  TerminalImage,
+  TerminalImageField,
+  TerminalImageCharacter,
+} from '@/utils/interfaces/3270Terminal';
+
+// Return a 2D array of characters, with each character representing all properties of its parent TerminalImageField.
+export default function getArrayOfImageCharacters(
+  imageData: TerminalImage
+): (TerminalImageCharacter | null)[][] {
+  const rows = imageData.imageSize.rows;
+  const columns = imageData.imageSize.columns;
+  // Initialise array with nulls.
+  const characterArray: (TerminalImageCharacter | null)[][] = Array.from({ length: rows }, () =>
+    Array(columns).fill(null)
+  );
+
+  imageData.fields.forEach(function (imageField: TerminalImageField) {
```

Comment:
Can the function in this `forEach` loop be a named function to make it easier to read?

If you think it can be split up into smaller functions, that would be even better!